iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -49,23 +49,23 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
     if (!maybeSegmentManager.has_value())
     {
         LogFatal() << "Invalid state! Could not obtain SegmentManager!";
-        std::terminate();
+        errorHandler(Error::kROUDI__PRECONDITIONS_FOR_PROCESS_MANAGER_NOT_FULFILLED, nullptr, ErrorLevel::FATAL);
```

Comment:
I thought every error enum should be used only once but here we use the same error enum at three places. 
The reason is that if we encounter an error value later in some log we exactly know where it originated, in this case however we would have to look at three possibilities.

Reply:
This was done intentionally to show there is still some work todo with our [RouDi/POSH split](https://github.com/eclipse-iceoryx/iceoryx/issues/539). Added `@todo` and moved to bottom.

Reply:
Sorry for opening this conversation again but I do not understand this. We intentionally create technical debt and dismiss our error handling concept (with unique error enums) so that we can add it to an issue and clean it up again?
Wouldn't it be easier to use three different error enums here right away and show that there is some work todo in the issue itself? Why do we have to show that there is work to do with bad code style?